pisek/task_jobs/testing_log.py

```diff
@@ -48,23 +52,30 @@ def _evaluate(self) -> None:
 
             solution = name[len(SOLUTION_MAN_CODE) :]
             solutions.add(solution)
-            log[solution] = {"results": []}
+            log["solutions"][solution] = {"results": {}}
+            solution_results = log["solutions"][solution]["results"]
 
             inp: TaskPath
-            sol_res: SolutionResult
+            sol_res: Optional[SolutionResult]
             for inp, sol_res in data["results"].items():
                 if sol_res is None:
                     warn_skipped = True
                     continue
-                log[solution]["results"].append(
-                    {
-                        "time": sol_res.solution_rr.time,
-                        "wall_clock_time": sol_res.solution_rr.wall_time,
-                        "test": inp.name,
-                        "points": sol_res.points,
-                        "result": sol_res.verdict.name,
-                    }
-                )
+                solution_results[inp.name] = {
+                    "time": sol_res.solution_rr.time,
+                    "wall_clock_time": sol_res.solution_rr.wall_time,
+                    "relative_points": (
+                        str(sol_res.relative_points)
+                        if isinstance(sol_res, RelativeSolutionResult)
+                        else None
+                    ),
+                    "absolute_points": (
+                        str(sol_res.absolute_points)
+                        if isinstance(sol_res, AbsoluteSolutionResult)
+                        else None
+                    ),
```

Comment:
I think it would be better to omit the other key entirely instead of setting it to `null`.